docs/proposals/scheduler/agent.proto

```diff
@@ -0,0 +1,46 @@
+syntax = "proto3";
+
+message Empty {}
+
+
+// ------------------
+// Lifecycle
+// ------------------
+
+message ConnectionRequest {
+    // agent certificate here
+}
+
+enum ConnectionStatus {
+    GRANTED = 0;
+    DENIED = 1;
+}
+
+message ConnectionResponse {
+    ConnectionStatus status = 1;
+}
+
+service LifecycleService {
+    rpc Connect(ConnectionRequest) returns (ConnectionResponse);
+    rpc Disconnect() returns ();
+}
+
+
+// ------------------
+// Status updates
+// ------------------
+
+message NodeStatusUpdateNotification {
+    // update about the node status
+    // such as amount of RAM, CPU, etc.
+}
+
+message InstanceStatusUpdateNotification {
```

Comment:
```suggestion
message InstanceStatus {
```

Also, you must agree on a terminology with the agent API. There they're proposing to use `Workload` to describe an instance. I suggest to keep `Workload` to describe a workload (its image, its limits, etc), and `Instance` for describing the instantiation of a `Workload`. There could be multiple `Instance`s of the same `Workload`, and an `Instance` exists when it's created on a node.

Reply:
Done. We only accept status updates for the `Instance`s, as Orka already knows stuff such as the `Workload` image, resource limits (I assume that the node agent won't change those, and that they are defined at the cluster level)...
The overall status of `Workload`s can also be retrieved via the status of its `Instance`s, so I don't see what information left the agent could provide us about `Workload`s that we don't already have elsewhere.